filters/src/passthrough.cpp

```diff
@@ -217,7 +217,7 @@ pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
         }
 
         // Unoptimized memcpys: assume fields x, y, z are in random order
-        memcpy (&pt[0], &input_->data[xyz_offset[0]], sizeof(float));
+        memcpy (pt.data(), &input_->data[xyz_offset[0]], sizeof(float));
```

Comment:
Having `pt.data()` for the first one and `&pt[1/2]` for the others is kind of strange. Can you also check the other files for similar occurrences, please?